ai_spider/stats.py

```diff
@@ -6,30 +6,109 @@
 from threading import Thread
 
 from ai_spider.unique_queue import UniqueQueue
+from database.schema import StatsContainer as StatsContainerTable, StatsWorker as StatsWorkerTable, StatsBin as StatsBinTable
+from sqlalchemy import and_
 
 STATS_EMA_ALPHA = 0.9
 PUNISH_SECS = 60 * 15
 PUNISH_BAD_PERF = 9999
 
+class StatsStore:
+    def __init__(self, session=None):
+        self.update_queue = UniqueQueue(key=lambda el: el[0])
+        self.update_thread = Thread(daemon=True, target=self.update_loop)
+        self.update_thread.start()
+        self.session=session
+
+    def stop(self, join=True):
+        self.update_queue.put([None, None])
+        if join:
+            self.update_thread.join()
+
+    def update_loop(self):
+        while True:
+            try:
+                key, val = self.update_queue.get(timeout=4)
+                if key is None:
+                    break
+                self._update(key, val)
+            except Empty:
+                pass
+
+    def update(self, key: str, val: dict):
+        self.update_queue.put([key, val])
+
+    def _update(self, key, val):
```

Comment:
leave this as an abstract base class, and create a new class called MySqlStatsStore